examples/user_interfaces/embed_in_paraview_sgskip.py

```diff
@@ -0,0 +1,103 @@
+"""
+===================================
+Embedding in a Paraview application
+===================================
+
+
+`ParaView <https://www.paraview.org/>`__ is a powerful 3D
+visualization tool.  It has both a Qt client application and python
+integration.
+
+This example shows how to embed an interactive Matplotlib figure into
+the Qt UI.
+
+ParaView (via VTK) embeds a python interpreter into the primary
+application.  This code is intended to be imported into the
+interpreter running within ParaView, find the main window and
+inject a dock widget with a Matplotlib Figure in it.  To use this,
+in the ParaView python shell ::
+
+ >> import sys
+ >> sys.path.append(path_to_examples)
+ >> import embed_in_paraview_sgksip as eip
+ >> fig = eip.make_dock_figure()
+ >> ax = fig.add_subplot(111)  # or ax = fig.subplots()
+ >> ln, = ax.plot(range(5))
+
+"""
+
+
+from PyQt5 import QtWidgets, QtCore
+from matplotlib.figure import Figure
+from matplotlib.backends.backend_qt5agg import FigureCanvas, NavigationToolbar2QT
+
+
+def get_main_window(target_name="pqClientMainWindow"):
+    """Find the main window
+
+    This assumes that the object name on the main window (where we
+    will be able to add a dock widget) has the object name
+
+    Parameters
+    ----------
+    target_name : str
+        The same of the MainWindow object of interest
+
+    Returns
+    -------
+    QtWidgets.QMainWindow
+
+    """
+    app = QtWidgets.QApplication.instance()
+    if app is None:
+        raise RuntimeError(
+            "No running QApplication found. " "May not be running inside of ParaView"
+        )
+
+    return next(
+        iter(
+            w
+            for w in app.topLevelWidgets()
+            if (
+                isinstance(w, QtWidgets.QMainWindow)
+                and w.objectName() == "pqClientMainWindow"
+            )
+        )
+    )
```

Comment:
I realize that *black* is the new black but... ugh :(
Also:
- you don't need the call to iter(), I think
- I guess you could skip the isinstance check? (it's unlikely to be a problem in reality) or use https://doc.qt.io/qt-5/qguiapplication.html#topLevelWindows?
- can you just use https://doc.qt.io/qt-5/qobject.html#findChild? (not sure it works on QApplication)